### ISSUE TITLE
Request for graceful rule export to file system

### ISSUE THREAD

**Issue by deterrak** [StackStorm/st2#3827]
# Requesting an elegant way to export a rule from the operational environment to the file system.
# It looks like the user would need to delete the 'extra' stuff (id, ref, uid...) in order to save the rule output to a file?  

root@u16:/opt/stackstorm/packs/default/rules# st2 rule get -y core.beagleTag
action:
    parameters:
        description: '{{trigger.body.description}}'
        gpio: '{{trigger.body.gpio}}'
        hostname: '{{trigger.body.hostname}}'
        ipaddress: '{{trigger.body.ipaddress}}'
        password: $password$
        trigger: '{{trigger.body.trigger}}'
        username: user
        value: '{{trigger.body.value}}'
    ref: default.beagleTagEnableNetworking
criteria: {}
description: Start the BeagleTag Workflow
enabled: true
**id: 59a46f89de59a4051d69096d**
name: beagleTag
pack: core
ref: core.beagleTag
tags: []
trigger:
    parameters:
        url: beagleTag
    **ref: core.c9603188-8c7a-41fd-9738-94cad5e90138**
    type: core.st2.webhook
type:
    parameters: {}
    ref: standard
**uid: rule:core:beagleTag**
 
root@u16:/opt/stackstorm/packs/default/rules#

**Comment by DavidJohnGee** [comment]
This comes up a lot in training and with both open source consumers and enterprise customers.

A feature to save rules back to the pack on disk, minus the ID/UID information would be great.

@coreteam: thoughts?

**Comment by LindsayHill** [comment]
Yes, agreed that this should be better. I would like to be able to do this from the UI - e.g. "sync to disk" - there's a similar issue related to workflow metadata for workflows created using Workflow Designer.
